File: microservices/product-service/src/models/product.js
const mongoose = require("mongoose");
const { Schema } = mongoose;

const ProductSchema = new Schema(
  {
    productName: { type: String },
    image: { type: String },
    nutrients: { type: String },
    from: { type: String },
    quantity: { type: String, default: "0" },
    price: { type: Number },
    organic: { type: String },
    description: { type: String }
  },
  {
    timestamps: { createdAt: true, updatedAt: true }
  }
);

module.exports = mongoose.model("Product", ProductSchema);

File: microservices/listing-service/Models/Product.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace listing_service.Models
{
    public class Product : IProduct
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
        [BsonElement("productName")]
        public string ProductName { get; set; }
        [BsonElement("image")]
        public string Image { get; set; }
        [BsonElement("from")]
        public string From { get; set; }
        [BsonElement("nutrients")]
        public string Nutrients { get; set; }
        [BsonElement("quantity")]
        public string Quantity { get; set; }
        [BsonElement("price")]
        public double Price { get; set; }
        [BsonElement("organic")]
        public bool Organic { get; set; }
        [BsonElement("description")]
        public string Description { get; set; }
    }
}